From 853fcd2329b427788677d11431e6165d108f935b Mon Sep 17 00:00:00 2001
From: Matt Hsu <matt_hsu@openmoko.org>
Date: Wed, 17 Sep 2008 16:10:43 +0100
Subject: [PATCH] Subject: [PATCH] rework-to-make-USBINS-USBREM-exclusive.patch

This patch is a work-around solution to correct charging indication logic.

Signed-off-by: Matt Hsu <matt_hsu@openmoko.org>
---
 drivers/i2c/chips/pcf50633.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/drivers/i2c/chips/pcf50633.c b/drivers/i2c/chips/pcf50633.c
index bac307e..c186fdb 100644
--- a/drivers/i2c/chips/pcf50633.c
+++ b/drivers/i2c/chips/pcf50633.c
@@ -915,7 +915,11 @@ static void pcf50633_work(struct work_struct *work)
 		add_request_to_adc_queue(pcf, PCF50633_ADCC1_MUX_ADCIN1,
 				     PCF50633_ADCC1_AVERAGE_16);
 	}
-	if (pcfirq[0] & PCF50633_INT1_USBREM) {
+	if (pcfirq[0] & PCF50633_INT1_USBREM &&
+				!(pcfirq[0] & PCF50633_INT1_USBINS)) {
+		/* the occurrence of USBINS and USBREM
+		 * should be exclusive in one schedule work
+		 */
 		DEBUGPC("USBREM ");
 
 		pcf->usb_removal_count++;
-- 
1.5.6.5

